frontend/simple/js/state.js

```diff
@@ -118,7 +171,7 @@ const actions = {
 
     const type = entry.type
     entry = Events[type].fromObject(entry, hash)
-    if (entry instanceof Events.HashableContract) {
+    if (entry instanceof Events.HashableContract && !state.contracts[contractId]) {
```

Comment:
So there is another case that also makes this needed and not skipping the first. If we skip the first as in most cases it would leave out the case of when we are joining a group for the first time (in which we need all events inclusive). I can make a special case for this. I think we should change the `since`  to `from` this way the name will suggest inclusion.